spec/unit/room.spec.ts

```diff
@@ -3225,4 +3226,52 @@ describe("Room", function () {
             expect(room.getBlacklistUnverifiedDevices()).toBe(false);
         });
     });
+
+    describe("findPredecessorRoomId", () => {
+        function roomCreateEvent(newRoomId: string, predecessorRoomId: string | null): MatrixEvent {
+            const content: {
+                creator: string;
+                ["m.federate"]: boolean;
+                room_version: string;
+                predecessor: { event_id: string; room_id: string } | undefined;
+            } = {
+                "creator": "@daryl:alexandria.example.com",
+                "predecessor": undefined,
+                "m.federate": true,
+                "room_version": "9",
+            };
+            if (predecessorRoomId) {
+                content.predecessor = {
+                    event_id: "spec_is_not_clear_what_id_this_is",
```

Comment:
For the record: the spec says "The event ID of the last known event in the old room.". Generally it's the tombstone event.

Reply:
Thank you! Where did you find that? I couldn't see much info in the Client-Server API spec for room upgrades.

Reply:
on the spec for `m.room.create` (https://spec.matrix.org/v1.5/client-server-api/#mroomcreate)